### ISSUE TITLE
Allow closing VS do NOT close modals by clicking outside on the background

### ISSUE THREAD

**Issue by imfx77** [kanboard/kanboard#5395]
### Configuration
- Kanboard version: 1.2.33
- Database type and version: sqlite 3.41.2 
- PHP version:  8.2.10 
- OS: Linux 4.4.302+ x86_64 (docker image on Synology)
- Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:121.0) Gecko/20100101 Firefox/121.0

### Related Issues (not in specific order and probably not a complete list):
https://github.com/kanboard/kanboard/issues/3791
https://github.com/kanboard/kanboard/issues/5288
https://github.com/kanboard/kanboard/pull/5363
https://github.com/kanboard/kanboard/pull/5226
https://github.com/kanboard/kanboard/pull/5258
https://github.com/kanboard/kanboard/issues/5239
https://github.com/kanboard/kanboard/issues/4895
https://github.com/kanboard/kanboard/issues/2153

I looked at those issues ... and I laughed at first, then I was sad, and lately I was angry!
Switch on, switch off, enable, disable ... 👿 ???

### The body of the problem
Allow closing modals by clicking on the background was introduced in 1.2.28, which I found very convenient.
It went through some modifications and bugfixes for several versions.
Then in 1.2.33 the feature got entirely disabled, as this appeares to be the easiest solution to some specific issue.
Hence, my frustration on the matter and the entire investigation about it.

I tagged this issue as **bug** but it is more of a **workflow** approach inconsistency.
I really don't mean to be rude, but you can't seriously run a software development like that!
That KILLS both the behavior consistency and the user experience!
You know, with features like that it is always 50:50, half the users will like it - the other half will hate it.
Or even: today they like it - tomorrow they hate it.
The wise approach in such situations is not to choose having it or not - but rather make it an option and let the users consider using it or not.

Anyway, regarding the specific matter - I don't even think there is currently a generic problem with the feature - I just blame it on an unfortunate sequence of issues, requests and decisions.
My personal preference is to have the option to click outside of a modal form so to discard it. There are multiple reasons why I like this better - habits, working with other platforms that behave this way, convenience, easier not having to aim with mouse for the small 'x' button or to hit the Esc key, not to mention the mobile experience where keyboard shortcuts don't help at all. 

As far as I understand, the guys who object this behavior don't really mind it but they rather face side d/effects it causes due to some bugs closing the modal inappropriately. What I see now is that :
* The 'dirtyForm' mechanic prevents closing a modal when there are changes, which is perfectly fine and works OK.
* The bugs about drag and select that end outside the modal and cause a fake 'click' event are now (almost) fixed.

I enabled locally for myself the 'overlayClickDestroy' option in all places in code and it seems to behave alright.
There is still a case when dragging from Tags field outside the Edit Task Modal form causes it to close inadequately.
But the point is - this is a bug - and bugs are to be fixed!
Masking bugs by disabling an entire feature is a BAD decision - especially when made authoritatively without any inquiry or discussion or gathering additional info. Removing something that works and was there for a while can't be expected to please the people who are already making use of it.

So, to wrap the whole thing:
* Please, revive the mechanic to close modals by clicking outside - yet consider making it an UI option, and definitely do hunt and kill the related bugs.
* I know this is an open source project, and mostly a one-man-army supported - yet, if you still aim at providing the best functionality and experience for the users - please don't sacrifice gathering different opinions and views in order to consider the best solution, rather than simply reacting on the fly to a particular request.

That being said, I am a long time user and fan of Kanboard ... and I don't wanna change that just now.
I really appreciate and THANK YOU for your continuous efforts, so please, take all the above on a good note!
Regards 🙇

**Comment by nekohayo** [comment]
Your complaint sounds like https://imgs.xkcd.com/comics/workflow.png

What is the difficulty with clicking the "Save" or "Cancel" (or "Close") buttons, really?

In my view, the previous approach was just unexpected and downright dangerous. If you really must have a shortcut to avoid clicking the Cancel / close button, then I think it would make more sense to bind the `Esc` keyboard shortcut to dismissing modals _when there are no unsaved changes,_ instead of any click outside doing that. Less risk of accidents (because yes, accidental clicks or touch presses happen, _all the time._ )

> consider making it an UI option

I wouldn't recommend this approach. Adding an "option" for this is non-design, that just complexifies the code and testing matrix, like the examples in https://www.joelonsoftware.com/2000/04/12/choices/

**Comment by rautamiekka** [comment]
> Less risk of accidents (because yes, accidental clicks or touch presses happen, all the time, for many users).

I can attest to that. Especially bad on Android and if the computer can't process the browser in a reasonable time, something, for ex, my Orange Pi 800 can't do with Firefox.

**Comment by tomeli5n** [comment]
Hello, I want to add a user-case for working with emojis using MarkdownPlus. 

- New Task modal
- user write Description text
- user write : 
- emoji selector pops up
- user press [ESC] key

expected behaviour: close emoji selector

actual behaviour: the New Task modal is gone


![image](https://github.com/kanboard/kanboard/assets/7417323/3cc720db-5bb2-491c-a834-2b0218b22138)
